homeassistant/components/tplink/fan.py

```diff
@@ -48,25 +53,26 @@ async def async_setup_entry(
 SPEED_RANGE = (1, 4)  # off is not included
 
 
-class TPLinkFan(CoordinatedTPLinkEntity, FanEntity):
+class Fan(CoordinatedTPLinkEntity, FanEntity):
     """Representation of a fan for a TPLink Fan device."""
 
     _attr_speed_count = int_states_in_range(SPEED_RANGE)
     _attr_supported_features = FanEntityFeature.SET_SPEED
+    _attr_has_entity_name = True
 
     def __init__(
         self,
         device: Device,
         coordinator: TPLinkDataUpdateCoordinator,
-        fan_module: Fan,
+        fan_module: FanInterface,
         parent: Device | None = None,
     ) -> None:
         """Initialize the fan."""
+        self._parent = parent
         super().__init__(device, coordinator, parent=parent)
```

Comment:
This reads odd, when the `super()` is called with `parent` directly after the statement, indicating that this should probably be done on the base class.

Reply:
Fixed, wasn't needed